pennylane/operation.py

```diff
@@ -235,11 +235,33 @@ def classproperty(func):
 
 
 def _process_data(op):
+
+    period = None
+
     if op.name in ("RX", "RY", "RZ", "PhaseShift", "Rot"):
-        return str([d % (2 * np.pi) for d in op.data])
+        period = 2 * np.pi
+
+    elif op.name in ("CRX", "CRY", "CRZ", "CRot"):
+        period = 4 * np.pi
+
+    if period is not None:
+
+        par_string = []
+        for d in op.data:
+
+            # If we got complex inputs for some reason, take the real part
+            # only. This can happen for example when differentiating
+            # holomorphic functions with JAX: e.g., a complex valued QNode (one
+            # that returns the state) requires complex typed inputs.
+            try:
+                complex_param = hasattr(d, "real")
+            except RuntimeError:
+                # Torch tensors raise an error
+                complex_param = False
+
+            par_string.append(d % period if not complex_param else d.real % period)
```

Comment:
@antalszava I wonder, could this be simplified down to just

```python
if op.name in ("RX", "RY", "RZ", "PhaseShift", "Rot"):
    return str([qml.math.real(d) % (2 * np.pi) for d in op.data])
```

This will ensure that it works with TF and autograd as well --- I have a feeling those also complained 🤔

Reply:
Yes! :heart_eyes: This works awesome with Torch and JAX too (those two cases I've seen have issues in the PR), changing to this. Thank you :slightly_smiling_face: 

With autograd, the jacobian of complex valued functions cannot be extracted it seems :confused: That's the case where they seem to divide complex numbers into 2 floats and we get a shape error.

Not sure about TF.